code/routers/user.js

```diff
@@ -0,0 +1,312 @@
+const express = require('express')
+const router = express.Router();
+const passport = require('passport');
+
+const User = require('../models/user');
+
+var async = require("async");
+var nodemailer = require("nodemailer");
+var crypto = require("crypto");
+
+
+
+/*
+  login api
+
+  required form inputs:
+      username: text
+      password: text ( password )
+
+  logs a user in using passport and creates local session
+  redirects user to /success or /failure depending on if the login was successful
+*/
+router.post('/login',
+  passport.authenticate('local', { successRedirect: '/success',
+                                   failureRedirect: '/failure'})
+);
+
+/*
+  signup api
+
+  required form inputs:
+      username: text
+      firstName: text
+      lastName: text
+      email: text
+      password: text ( password )
+      verifyPassword: text ( password )
+
+  checks given information and creates a new user
+*/
+router.post('/signup', function(req, res) {
+  const { username, firstName, lastName, email, password, verifyPassword } = req.body;
+
+  // check if required fields are filled
+  // ( check if there is a better way of checking these )
+  if (!username || !firstName || !lastName || !email || !password || !verifyPassword) {
+    return res.status(400).send({
+      message: 'Fill in required fields'
+    });
+  }
+
+  // verify password
+  if ( password != verifyPassword ) {
+    return res.status(400).send({
+      message: 'Passwords do not match'
+    });
+  }
+
+  // password requirements
+  if ( !/[a-z]/.test(password) || // check for lowercase letter
+       !/[A-Z]/.test(password) || // check for uppercase letter
+       password.length < 8 ){ // check if password has atleast 8 letters
+
+    return res.status(400).send({
+      message: 'Password requirements not filled'
+    });
+  }
+
+  // check if user already exists
+  let user_query = {$or: [
+    {username: username},
+    {email: username}
+  ]}
+
+  User.findOne(user_query, function(err, user, next) {
+    if (user) {
+      return res.status(400).send({
+        message: 'User already exists with that username or email'
+      });
+    }else{
+      // create user
+      let newUser = new User();
+
+      newUser.username = username;
+      newUser.firstName = firstName;
+      newUser.lastName = lastName;
+      newUser.email = email;
+      newUser.setPassword(password);
+
+      newUser.save(function(err, user_check) {
+        if(err) {
+          return res.status(400).send({
+            message: 'Failed to add user'
+          });
+        }
+        else {
+          return res.status(201).send({
+            message: 'User added successfully'
+          });
+        }
+      });
+    }
+  });
+
+});
+
+
+/*
+  forgot password api
+
+  let's users reset their password and redirect
+  them to the login page
+*/
+
+router.get('/forgot', function(req, res) {
```

Comment:
I don't think we need this anymore since react will now be covering which page we are on. I may be misunderstanding what this is though. It is for redirecting the user the 'forgot' page right? If this is just for redirecting I don't think we need it anymore.